File: ml_js/src/commons/utils/transporter/websocket_transporter.js
import Transporter from './transporter';
import {MLPyHost, MLPyPort} from "../../settings";

export default class WebsocketTransporter extends Transporter {
    constructor(project_id, call_back, trace_id) {
        super(project_id, call_back, trace_id);

        this.mlPyUrl = `ws://${MLPyHost}:${MLPyPort}/ws/${project_id}`;
        this.socket = new WebSocket(this.mlPyUrl);
    }

    init() {
        this.socket.onmessage = (event) => {
            const data = JSON.parse(event.data);
            this.received(data);
        };

        this.socket.onclose = () => {
            console.error('Chat socket closed unexpectedly')
        };
    }

    send(data) {
        if (data.action === 'listen') return;
        data['trace_id'] = this.trace_id;
        console.log(data);
        this.socket.send(JSON.stringify(data));
    }

}

File: ml_py/lib/transporter/websocket_transporter.py
from lib.transporter import Transporter
import json


class WebsocketTransporter(Transporter):
    def __init__(self, project_id, trace_id=None, consumer=None):
        super().__init__(project_id, trace_id)
        self.consumer = consumer

    def send(self, data):
        self.set_trace_id()
        data['trace_id'] = self.trace_id
        self.consumer.send(text_data=json.dumps(data))

